### PR TITLE
Fix TypeError: Cannot read properties of undefined (reading 'treatAsCategorical')"

### PR DESCRIPTION
**Pull request by vinuthakaranth**
## Description

Fix TypeError: Cannot read properties of undefined (reading 'treatAsCategorical')".
metaDict[xKey] can be undefined leading to this error.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] Documentation was updated if it was needed.
- [ ] New tests were added or changes were manually verified.

### CODE REVIEW

**Review by tongyu-microsoft — APPROVED** [review]

File: libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx [hunk 1]
```diff
@@ -438,7 +438,7 @@ export class GlobalExplanationTab extends React.PureComponent<
     // set to dependence plot initially, can be changed if other feature importances available
     const xKey = JointDataset.DataLabelRoot + featureIndex.toString();
     const xIsDithered =
-      this.context.jointDataset.metaDict[xKey].treatAsCategorical;
+      this.context.jointDataset.metaDict[xKey]?.treatAsCategorical;
```
Comment (gaugup) [inline libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: Is there a unit test for this change? In what situations can treatAsCategorical be not available as a property?

**Review by imatiach-msft — APPROVED** [review]

File: libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx (hunk 1 above)
Comment (vinuthakaranth) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: In this case when xKey isn't available, treatAsCategorical will be undefined. @imatiach-msft [imatiach-msft] or @riedgar-ms  any idea when is xKey not available?

Comment (imatiach-msft) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: hmm, no idea.  It's unfortunate we don't have a repro of this issue...

**Review by romanlutz — APPROVED** [review]

File: libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx (hunk 1 above)
Comment (riedgar-ms) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: Is there any logging we can usefully do for when the 'null' case is hit? Or are the usual values of xKey considered customer data?

Comment (gaugup) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: are there other places in the code that we need to fix knowing that `treatAsCategorical` can be optional?

**Review by gaugup — APPROVED** [review]

File: libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx (hunk 1 above)
Comment (gaugup) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: Thanks for fixing this everywhere.

Comment (vinuthakaranth) [reply libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx:441]: Thanks for catching and commenting about this!! :)

**Pull request closed without merging**